server/src/main/java/io/crate/expression/symbol/WindowFunction.java

```diff
@@ -80,6 +81,36 @@ public Boolean ignoreNulls() {
         return ignoreNulls;
     }
 
+    @Override
+    public boolean any(Predicate<? super Symbol> predicate) {
```

Comment:
What's not so nice about this change is that these `any` implementations now duplicate logic in `DefaultTraversalSymbolVisitor`

I suspect we could either implement `any` in terms of a `DefaultTraversalSymbolVisitor`, or we could go the other way around, and try to implement more operations in terms of `any`. Given that it visits each node in a tree that should work - and it even has short-circuit capabilities. E.g. to replace `RefVisitor` we could have:

```java
    default void visitRefs(Consumer<? super Reference> consumer) {
        any(s -> {
            if (s instanceof Reference ref) {
                consumer.accept(ref);
            }
            return false;
        });
    }
```